### PR TITLE
Converted more classes in scene folder to use let and const

### PR DESCRIPTION
**Pull request by mvaligursky**
- conversions of var -> let / const
- split one large function in batch-manager to two

### CODE REVIEW

File: src/scene/graph-node.js [hunk 1]
```diff
@@ -6,21 +6,21 @@ import { Mat4 } from '../math/mat4.js';
...
+const scaleCompensateScaleForParent = new Vec3();
+const tmpMat4 = new Mat4();
+const tmpQuat = new Quat();
+const position = new Vec3();
+const invParentWtm = new Mat4();
+const rotation = new Quat();
+const invParentRot = new Quat();
+const matrix = new Mat4();
+const target = new Vec3();
+const up = new Vec3();
```
Comment (yaustar) [inline src/scene/graph-node.js:23]: Shouldn't these technically be `let` as they will change value?

**Review by yaustar — APPROVED** [review]
LGTM with comment

File: src/scene/graph-node.js (hunk 1 above)
Comment (mvaligursky) [reply src/scene/graph-node.js:23]: it's not c++ or c# type of const.
In JS const is if you don't assign anything else to it. So you can even have `const array = [];` and add and remove elements from it. As long as you don't do `array = null;` or similar

Comment (yaustar) [reply src/scene/graph-node.js:23]: Oh, I see. JS is weird 😅

**Pull request merged**